cloudinit/sources/DataSourceMAAS.py

```diff
@@ -164,6 +173,22 @@ def check_instance_id(self, sys_cfg):
         return self.id_hash == get_id_from_ds_cfg(ncfg)
 
 
+class DataSourceMAASLocal(DataSourceMAAS):
+    network_config_sources = (
+        NetworkConfigSource.CMD_LINE,
+        NetworkConfigSource.SYSTEM_CFG,
+        NetworkConfigSource.DS,
+        NetworkConfigSource.INITRAMFS,
+    )
+
+    def _get_data(self):
+        if not KlibcNetworkConfigSource().is_applicable():
+            LOG.debug("No initramfs applicable config")
+            return False
+        LOG.debug("Found initramfs applicable config")
```

Comment:
So if there's no initramfs network configuration, we're not going to try to configure network at all? Why is that?

Reply:
@TheRealFalcon The return True/False here is just whether the DataSourceMAASLocal is being detected in init-local timeframe, not whether network is  configured by the datasource.


It means that we are going `return False` and avoid detecting DataSourceMAASLocal when there is no applicable initramfs ntework config available in init-local stage. In absence of initramfs network config, cloud-init will fallback to detecting DataSourceMAAS in init-network boot stage and rely on the typical priority of network config sources which still prioritize the typical priority order of network_config_sources.

Reply:
I'll add context in comments inline.